### ISSUE TITLE
File picker: ignore git submodules

### ISSUE THREAD

**Issue by beeb** [helix-editor/helix#9076]
The [file picker section](https://docs.helix-editor.com/configuration.html?highlight=file-picker#editorfile-picker-section) of the config provides much-useful options to disable showing certain types of hidden or git files in the file picker.

However, I would love to have an option to disable any file located inside a git submodule. Take for instance [foundry](https://github.com/foundry-rs/foundry), which uses git submodules for project libraries/dependencies. When I add e.g. `forge-std` as a git submodule, which gets cloned into the `lib` folder, then all the source files for `forge-std` show up in my file picker. 

I couldn't find an easy way to hide those, as obviously I can't git-ignore a git submodule.

Thanks!

**Comment by kirawi** [comment]
You can use a `.helix/ignore` file

**Comment by pascalkuthe** [comment]
Yeah custom ignore logic like that is hard to add because it's not supported by the ignore crate. I also think while this may be useful in this one generic instance it doesn't seem generally useful (or a common ignore convention). Using .helix/ignore seems more appropriate
